Alpakka 2.0 (#703)
akka/akka.io: blog/_posts/2020-04-30-alpakka-2.0.0-released.md

---
layout: post
title: Alpakka 2.0 Released
author: Enno
short: Please meet Alpakka 2.0
category: news
tags: [releases]
---

Dear Hakkers,

The Alpakka contributors are happy to announce **Alpakka 2.0**!

## What is Alpakka?

The [Alpakka project](https://doc.akka.io/docs/alpakka/current/) is an open source initiative to implement stream-aware and reactive integration pipelines for Java and Scala. It is built on top of [Akka Streams](https://doc.akka.io/docs/akka/current/stream/index.html), and has been designed from the ground up to understand streaming natively and provide a DSL for reactive and stream-oriented programming, with built-in support for backpressure. Akka Streams is a [Reactive Streams](https://www.reactive-streams.org/) and JDK 9+ [java.util.concurrent.Flow](https://docs.oracle.com/javase/10/docs/api/java/util/concurrent/Flow.html)-compliant implementation and therefore [fully interoperable](https://doc.akka.io/docs/akka/current/general/stream/stream-design.html#interoperation-with-other-reactive-streams-implementations) with other implementations.

## About 2.0

> Things have taken much longer than we initially hoped and we hope the earlier milestone releases made it possible for the impatient users of Alpakka to try out the current versions and fixes.

Reviewer: do we need to excuse ourselves?

> Things have taken much longer than we initially hoped and we hope the earlier milestone releases made it possible for the impatient users of Alpakka to try out the current versions and fixes.

Reviewer 2: ```suggestion
Things have taken much longer than we initially planned, but we hope the earlier milestone releases made it possible for our early adopters of Alpakka to try out the current versions and fixes.
```

The major version tick for Alpakka is due to major upgrades and changes in the dependencies used by the various connectors. Some Alpakka APIs changed as well, but we try to keep those changes minimal.

**Alpakka 2.0 updates dependencies for most connectors.**

> **Alpakka 2.0 requires Akka 2.5.31 or Akka 2.6.4.** We recommend to upgrade your projects to the latest versions of Akka. Akka stays [binary-compatible](https://doc.akka.io/docs/akka/current/common/binary-compatibility-rules.html#binary-compatibility-rules) between patch releases (and for the most part between 2.5.x and 2.6.x). 

Reviewer: 2.6.4 or later.

Both the classic and the new actors API can be used with Alpakka.

Future releases of Alpakka will require Akka 2.6.

Most Alpakka APIs have NOT changed since Alpakka 1.0. We hope adapting to the changes will be easy. For the time being we don't have dedicated migration guides. Please reach out in the [Discuss forum](https://discuss.lightbend.com/c/akka/streams-and-alpakka) if you run into problems.

## Major new functionality

- Pravega connector [#2149](https://github.com/akka/alpakka/issues/2149) by [@cheleb](https://github.com/cheleb)
   
    [Pravega](http://www.pravega.io/) provides a new storage abstraction - a stream - for continuous and unbounded data. A Pravega stream is an elastic set of durable and append-only segments, each segment being an unbounded sequence of bytes. Streams provide exactly-once semantics, and atomicity for groups of events using transactions.

- Cassandra: rewrite on top of Java driver 4.5.0 [#2182](https://github.com/akka/alpakka/issues/2182) by [@ennru](https://github.com/ennru)

>     **Alpakka Cassandra is completely rewritten and now based on the Cassandra Java driver version 4.6. It now becomes the base for [Akka Persistence Cassandra](https://doc.akka.io/docs/akka-persistence-cassandra/1.0/).**

Reviewer: in sentence above it says 4.5.0

- AWS Kinesis KCL streams support [#1667](https://github.com/akka/alpakka/issues/1667) by [@aserrallerios](https://github.com/aserrallerios)

- Elasticsearch
    - add support for version 7 [#2277](https://github.com/akka/alpakka/issues/2277) by [@r-glyde](https://github.com/r-glyde)

- File 
    - TAR archive generation [#2241](https://github.com/akka/alpakka/issues/2241) by [@choffmeister](https://github.com/choffmeister)
    - TAR archive reader [#2268](https://github.com/akka/alpakka/issues/2268) by [@ennru](https://github.com/ennru)
    - Added flow for creating file ZIP archive [#1890](https://github.com/akka/alpakka/pull/1890) by [@michalbogacz](https://github.com/michalbogacz)

We've adapted the Alpakka APIs even more to enable the use of the new actor APIs where possible.

Read the full [Release notes](https://doc.akka.io/docs/alpakka/2.0/release-notes/2.0.x.html) for all details.

## The way forward

> The Alpakka community is encouraged to continue to suggesting improvements and new functionality to the existing connectors, or even suggest integrations with other technologies. During Alpakka 2.0 we will keep the APIs binary-compatible.

Reviewer: "During Alpakka 2.0 patch releases" or 2.0.x

> The Alpakka community is encouraged to continue to suggesting improvements and new functionality to the existing connectors, or even suggest integrations with other technologies. Alpakka 2.0 patch releases we will keep the APIs binary-compatible.

Reviewer 2: ```suggestion
The Alpakka community is encouraged to continue to suggest improvements and new functionality to the existing connectors, or even suggest integrations with other technologies. Alpakka 2.0 patch releases we will keep the APIs binary-compatible.
```
